rpc/filters.go

```diff
@@ -12,32 +13,80 @@ import (
 	Used to set the criteria passed in from RPC params
 */
 
+// Backend implements the functionality needed to filter changes.
+// Implemented by PublicEthAPI.
+type Backend interface {
```

Comment:
What specifically is the reason you are creating this interface?

Reply:
so that it's clear what functions the Filter needs, also in case we want to switch out the backend later